generators/pkg/writer/write_test.go

```diff
@@ -0,0 +1,52 @@
+package writer
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestWriteToDisk(t *testing.T) {
+	task := v1beta1.Task{
+		Spec: v1beta1.TaskSpec{
+			Workspaces: []v1beta1.WorkspaceDeclaration{{Name: "input",
+				MountPath: "/input"}},
+			Steps: []v1beta1.Step{{Container: corev1.Container{
+				Name:    "build",
+				Image:   "gcr.io/kaniko-project/executor:latest",
+				Command: []string{"/kaniko/executor"},
+				Args: []string{"--context=dir://$(workspaces.input.path)/src",
+					"--destination=gcr.io/wlynch-test/kaniko-test",
+					"--verbosity=debug"},
+			}}}}}
+
+	b, err := WriteToDisk(task)
+	if err != nil {
+		t.Fatalf("error from 'Writer': %v", err)
+	}
+
+	got := string(b)
+	want := `
+	    metadata:
```

Comment:
The reason why you might have been seeing indent errors, is because anything inside `` is treated as a string literal, which means that any indenting you put it to make it look indented w.r.t. the code will also be included in the resulting string.

We need to do one of 2 things:
1. Remove the extra whitespace from the want string in the code here.
2. Move this to its own yaml file in a testdata dir and read it in.
These are functionally equivalent so up to you which one you want to do.

Also I'm confused why we're not seeing presubmit failures for this. We should file an issue to make sure PR testing is set up.

Reply:
Done. I filed the issue #566 here.